faroe/api/src/main/java/org/n52/faroe/controller/SettingsController.java

```diff
@@ -0,0 +1,162 @@
+            newSettingGenerator(newSettings, settingValueFactory);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error("Couldn't add new Settings");
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @DeleteMapping({"/{settingId}"})
+    public ResponseEntity<Object> deleteSettings(@PathVariable("service") String serviceName,
+                                                 @PathVariable("settingId") String settingId) {
+        LOG.info(String.format("Deleting setting id %s for service %s", settingId, serviceName));
+        try {
+            final SettingsService settingsService = servicesDao.getServiceByName(serviceName)
+                    .getSettingsService();
+            settingsService.deleteSetting(settingsService.getDefinitionByKey(settingId));
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error(String.format("Couldn't delete settings with the settings id %s", settingId));
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping({"/{definitionId}"})
+    public ResponseEntity<Object> updateSettings(@PathVariable("service") String serviceName,
+                                                 @PathVariable("definitionId") String settingId,
+                                                 @RequestBody String newSettings) {
+        LOG.info(String.format("Updating Setting with id %s for service %s", settingId, serviceName));
+        try {
+            final JsonObject newSettingsJson = new JsonParser().parse(newSettings).getAsJsonObject();
+            final SettingsService settingsService = servicesDao.getServiceByName(serviceName).getSettingsService();
+            final SettingValue updatedValue = newSettingGenerator(newSettingsJson,
+                    settingsService.getSettingFactory());
+            settingsService.changeSetting(updatedValue);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error(String.format("Couldn't update settings with setting id %s for service %s ",
+                        settingId, serviceName));
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * For PUT operations, the unchanged values will be sent as part of form input.
+     * For POST operations, all the values will need part of the input payload.
+     */
+    private SettingValue<?> newSettingGenerator(JsonObject newSettings, SettingValueFactory settingValueFactory) {
+        return Optional.ofNullable(newSettings.get(Constants.TYPE)).map(type -> {
+            Preconditions.checkNotNull(newSettings.get(Constants.KEY));
+            Preconditions.checkNotNull(newSettings.get(Constants.VALUE));
+            switch (type.getAsString().toLowerCase()) {
+                case "boolean":
```

Comment:
I think it's better to refactor `SettingsValueFactory.newSettingValue()` to also accept two strings. Duplicating the code here seems wrong...